Userland/Libraries/LibGfx/BitmapSequence.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024, Zachary Huang <zack466@gmail.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/AnonymousBuffer.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/BitmapSequence.h>
+#include <LibGfx/Size.h>
+#include <LibIPC/Decoder.h>
+#include <LibIPC/Encoder.h>
+#include <LibIPC/File.h>
+
+namespace Gfx {
+
+BitmapSequence::BitmapSequence(Vector<Optional<NonnullRefPtr<Bitmap>>> bitmaps)
+    : m_bitmaps(move(bitmaps))
+{
+}
+
+}
+
+namespace IPC {
+
+template<>
+ErrorOr<void> encode(Encoder& encoder, Gfx::BitmapSequence const& bitmap_sequence)
+{
+    Vector<Optional<NonnullRefPtr<Gfx::Bitmap>>> bitmaps = bitmap_sequence.bitmaps();
```

Comment:
even if bitmaps() returns a const&, using `auto const&` or `full type name const&` is better style here. if it doesn't return a const&, it should :)